### PR TITLE
Integrate with sdk

### PR DESCRIPTION
**Pull request by agarneha1331**

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1 +1,2 @@
 * @browserstack/app-automate-public-repos
+* @browserstack/automate-public-repos
```
Comment (07souravkunda) [inline CODEOWNERS:2]: this is showing error

File: android/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  "version": "0.1.0",
+  "readme": "NodeJS Integration with [BrowserStack](https://www.browserstack.com)",
+  "description": "Appium android examples for NodeJS and BrowserStack App Automate",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/browserstack/node-appium-app-browserstack.git"
+  },
+  "scripts": {
+    "sample-test": "browserstack-node-sdk node test/sample_test.js",
+    "sample-local-test": "browserstack-node-sdk node test/sample_local_test.js"
```
Comment (07souravkunda) [inline android/package.json:12]: add post install script, `npm update browserstack-node-sdk`

File: android/test/sample_local_test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    ).getText();
+
+    assert(textElement.includes('The active connection is wifi'));
+    assert(textElement.includes('Up and running'));
+
+    await driver.executeScript(
+      'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed","reason": "The connection is up and running"}}'
+    );
+  } catch (e) {
+    console.log(e);
```
Comment (07souravkunda) [inline android/test/sample_local_test.js:36]: remove console

File: android/test/sample_test.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        '/hierarchy/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.FrameLayout/android.widget.LinearLayout[1]/android.widget.FrameLayout[2]/android.widget.FrameLayout/android.widget.LinearLayout/android.widget.ListView/android.widget.LinearLayout'
+      )
+    );
+
+    assert(allProductsName.length > 0);
+    await driver.executeScript(
+      'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed","reason": "Search in Wikipedia done correctly"}}'
+    );
+  } catch (e) {
+    console.log(e);
```
Comment (07souravkunda) [inline android/test/sample_test.js:42]: remove console

File: ios/package.json [hunk 5]
```diff
@@ -7,6 +7,10 @@
     "type": "git",
     "url": "git+https://github.com/browserstack/node-appium-app-browserstack.git"
   },
+  "scripts": {
+    "sample-test": "browserstack-node-sdk node test/sample_test.js",
+    "sample-local-test": "browserstack-node-sdk node test/sample_local_test.js"
```
Comment (07souravkunda) [inline ios/package.json:12]: add postinstall script

File: ios/package.json [hunk 6]
```diff
@@ -17,12 +21,17 @@
...
   "homepage": "https://github.com/browserstack/node-appium-app-browserstack#readme",
   "dependencies": {
-    "browserstack-local": "^1.0.2",
-    "wd": "^1.13.0",
+    "browserstack-local": "^1.5.3",
     "chai": "^4.2.0",
-    "chai-as-promised": "7.1.1"
+    "chai-as-promised": "7.1.1",
+    "selenium-webdriver": "^4.10.0",
+    "wd": "^1.13.0"
```
Comment (07souravkunda) [inline ios/package.json:28]: wd and chai required?

File: ios/package.json [hunk 7]
```diff
@@ -17,12 +21,17 @@
...
   },
   "engines": {
     "node": "^6"
+  },
+  "devDependencies": {
+    "browserstack-node-sdk": "latest",
+    "mocha": "^10.2.0"
```
Comment (07souravkunda) [inline ios/package.json:35]: mocha in vanilla node?

File: ios/test/sample_local_test.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        '/XCUIElementTypeApplication/XCUIElementTypeWindow[1]/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeTextField'
+      )
+    ).getText();
+
+    assert(textElement.includes('Up and running'));
+    await driver.executeScript(
+      'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed","reason": "The connection is up and running"}}'
+    );
+  } catch (e) {
+    console.log(e);
```
Comment (07souravkunda) [inline ios/test/sample_local_test.js:32]: remove console

File: ios/test/sample_test.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        '/XCUIElementTypeApplication/XCUIElementTypeWindow[1]/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeOther/XCUIElementTypeStaticText'
+      )
+    ).getText();
+
+    assert(textOutput === 'hello@browserstack.com');
+    await driver.executeScript(
+      'browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed","reason": "Input and output text matches"}}'
+    );
+  } catch (e) {
+    console.log(e);
```
Comment (07souravkunda) [inline ios/test/sample_test.js:42]: remove console

**Review by 07souravkunda — CHANGES_REQUESTED** [review]

File: android/browserstack.yml [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+# =======================
+# Parallels per Platform
+# =======================
+# The number of parallel threads to be used for each platform set.
+# BrowserStack's SDK runner will select the best strategy based on the configured value
+#
+# Example 1 - If you have configured 3 platforms and set `parallelsPerPlatform` as 2, a total of 6 (2 * 3) parallel threads will be used on BrowserStack
+#
+# Example 2 - If you have configured 1 platform and set `parallelsPerPlatform` as 5, a total of 5 (1 * 5) parallel threads will be used on BrowserStack
+parallelsPerPlatform: 1
```
Comment (07souravkunda) [inline android/browserstack.yml:59]: remove ppp, as we don't support in case of vanilla

**Review by 07souravkunda — CHANGES_REQUESTED** [review]

File: android/browserstack.yml (hunk 10 above)
Comment (agarneha1331) [reply android/browserstack.yml]: done

**Review by 07souravkunda — APPROVED** [review]

**Review by deepwired — APPROVED** [review]
LGTM

**Pull request merged**